devicereg/devicereg.proto

```diff
@@ -97,13 +140,198 @@ message RevokeDeviceRequest {
   // The unique token identifying the device.
   string device_token = 1;
 
-  // The user's public key, serving here just to prove that the user actually is
+  // The device's private key, serving here just to prove that the user actually is
   // the entity that previously claimed the device.
-  string user_public_key = 2;
+  string device_private_key = 2;
 }
 
 // RevokeDeviceResponse is a placeholder response returned from a revoke
 // request. Currently empty, but reserved for any fields identified for future
 // iterations.
 message RevokeDeviceResponse {
 }
+
+// Entitlement is a message used to describe an operation that may be applied to
+// a specific data type published by a SmartCitizen device. The message contains
+// two required fields: the sensor_id (this is the type of data we are entitling
+// over), and a specified operation to be performed on that sensor type. This
+// can be one of three actions: to share the sensor without modification, to
+// apply a binning algorithm to the data so we output a bucketed value, or a
+// moving average calculated dynamically for incoming values.
+//
+// If an titlement specifies an Action type of `BIN`, then the optional
+// `buckets` parameter is required, similarly if an action type of `MOVING_AVG`
+// is specified, then `interval` is a required field.
+message Entitlement {
+  // The unique id of the sensor type for which this specific entitlement is
+  // defined. This is a required field.
+  uint32 sensor_id = 1;
+
+  // An enumeration which allows us to specify what type of sharing is to be
+  // defined for the specified sensor type. The default value is `SHARE` which
+  // implies sharing the data at full resolution. If this type is specified, it
+  // is an error if either of `buckets` or `interval` is also supplied.
+  enum Action {
+    SHARE = 0;
+    BIN = 1;
+    MOVING_AVG = 2;
+  }
+
+  // The specific action this entitlement defines for the sensor type. This is a
+  // required field.
+  Action action = 2;
+
+  // The buckets attribute is used to specify the the buckets or bins into which
+  // incoming values should be stored. Each element in the list is the upper
+  // inclusive bound of a bucket. The values submitted must be sorted in
+  // strictly increasing order. There is no need to add a highest bucket with
+  // +Inf bound, it will be added implicitly. This field is optional unless an
```

Comment:
I'm not entirely sure I understand what "Each element in the list is the upper inclusive bound of a bucket. The values submitted must be sorted in strictly increasing order." means